### ISSUE TITLE
Segfault with caching and parallel=True.

### ISSUE THREAD

**Issue by DrTodd13** [numba/numba#7518]
```
import numba
import numpy as np

@numba.njit(cache=True, parallel=True)
def test(a):
    b = a+5
    print(b.shape)
    return b

a = np.ones(1000)
b = test(a)
print(b.shape)
```

Run that with the following:
```
NUMBA_NUM_THREADS=8 python /tmp/caching1.py
NUMBA_NUM_THREADS=16 python /tmp/caching1.py
```

That results in a segfault.

**Comment by stuartarchibald** [comment]
@DrTodd13 was this resolved by #7625? If so, can it be closed? Thanks for checking.

**Comment by DrTodd13** [comment]
Confirmed that this issue is no longer present.
